test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/InstanceAnnotations/AnnotationController.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using Microsoft.AspNet.OData;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.Test.E2E.AspNet.OData.Common.Controllers;
+using Xunit;
+
+namespace Microsoft.Test.E2E.AspNet.OData.InstanceAnnotations
+{
+    public class EmployeesController : TestODataController
+    {
+        public EmployeesController()
+        {
+            if (null == Employees)
+            {
+                InitEmployees();
+            }
+        }
+
+        /// <summary>
+        /// static so that the data is shared among requests.
+        /// </summary>
+        private static IList<Employee> Employees = null;
+
+        private void InitEmployees()
+        {
+            Employees = new List<Employee>
+            {
+                new Employee()
+                {
+                    ID=1,
+                    Name="Name1",
+                    SkillSet=new List<Skill>{Skill.CSharp,Skill.Sql},
+                    Gender=Gender.Female,
+                    AccessLevel=AccessLevel.Execute,
+                    FavoriteSports=new FavoriteSports()
+                    {
+                        LikeMost=Sport.Pingpong,
+                        Like=new List<Sport>{Sport.Pingpong,Sport.Basketball}
+                    }
+                },
+                new Employee()
+                {
+                    ID=2,Name="Name2",
+                    SkillSet=new List<Skill>(),
+                    Gender=Gender.Female,
+                    AccessLevel=AccessLevel.Read,
+                    FavoriteSports=new FavoriteSports()
+                    {
+                        LikeMost=Sport.Pingpong,
+                        Like=new List<Sport>{Sport.Pingpong,Sport.Basketball}
+                    }
+                },
+                new Employee(){
+                    ID=3,Name="Name3",
+                    SkillSet=new List<Skill>{Skill.Web,Skill.Sql},
+                    Gender=Gender.Female,
+                    AccessLevel=AccessLevel.Read|AccessLevel.Write,
+                    FavoriteSports=new FavoriteSports()
+                    {
+                        LikeMost=Sport.Pingpong|Sport.Basketball,
+                        Like=new List<Sport>{Sport.Pingpong,Sport.Basketball}
+                    }
+                },
+            };
+
+            var instanceAnnot = new ODataInstanceAnnotationContainer();
+            instanceAnnot.AddResourceAnnotation("NS.Test2", 2);
```

Comment:
>"NS.Test2 [](start = 48, length = 9)

can you add another test to cover a hash (#) , for example:

"Orders@com.example.display.style#simple": { "order": 2 }

Reply:
AddResourceAnnotation("@NS.Test", 2)

---
In reply to: [490471905](https://github.com/OData/WebApi/pull/2219#discussion_r490471905) [](ancestors = 490471905)

Reply:
Curious what ODL does with "#" in the name.  If ODL doesn't preclude it, we should support (name = "com.example.display.style#simple")
As per above, the later ("@NS.Test") should fail.

---
In reply to: [490471905](https://github.com/OData/WebApi/pull/2219#discussion_r490471905) [](ancestors = 490471905)